src/components/studies/participants/DialogContents.tsx

```diff
@@ -0,0 +1,248 @@
+import React, { useEffect } from 'react'
+import { latoFont } from '../../../style/theme'
+import {
+  EnrollmentType,
+  ParticipantAccountSummary,
+  Study,
+  ParticipantActivityType,
+  EnrolledAccountRecord,
+} from '../../../types/types'
+import clsx from 'clsx'
+import ParticipantService from '../../../services/participants.service'
+
+const useStyles = makeStyles(theme => ({
+  root: {
+    width: '80%',
+    fontFamily: latoFont,
+    fontSize: '16px',
+  },
+  idList: {
+    textAlign: 'center',
+    fontWeight: 'bold',
+    marginBottom: theme.spacing(2.5),
+    marginTop: theme.spacing(2),
+    '& span': {
+      display: 'block',
+    },
+    maxHeight: '200px',
+    overflow: 'auto',
+  },
+  smsIdListContainer: {
+    backgroundColor: '#F8F8F8',
+    padding: theme.spacing(1, 1, 1, 2),
+    textAlign: 'left',
+    marginBottom: theme.spacing(3.5),
+  },
+}))
+
+type DialogContentsProps = {
+  participantsWithError: ParticipantAccountSummary[]
+  selectedParticipants: ParticipantAccountSummary[]
+  study: Study
+  isProcessing: boolean
+  // if this is false, then sms is implied to be true
+  isRemove: boolean
+  selectingAll: boolean
+  tab: ParticipantActivityType
+  token: string
+}
+
+function formatExternalID(id: string) {
+  const arr = id.split(':')
+  return arr.length === 1 ? arr[0] : arr[1]
+}
+
+function formatIds(
+  enrollmentType: EnrollmentType,
+  participants: EnrolledAccountRecord[],
+): string[] {
+  return participants.map((participant: EnrolledAccountRecord) =>
+    enrollmentType === 'PHONE'
+      ? participant.participant.phone?.nationalFormat ||
+        formatExternalID(participant.externalId) ||
+        'unknown'
+      : formatExternalID(participant.externalId) || 'unknown',
+  )
+}
+
+const DialogContents: React.FunctionComponent<DialogContentsProps> = ({
+  participantsWithError,
+  selectedParticipants,
+  study,
+  isProcessing,
+  isRemove,
+  selectingAll,
+  tab,
+  token,
+}) => {
+  const classes = useStyles()
+  const [participantData, setParticipantData] = React.useState<
+    EnrolledAccountRecord[]
+  >([])
+  const [loadingData, setLoadingData] = React.useState(false)
+
+  const setData = async () => {
+    const resultEnrolled = await ParticipantService.getAllParticipantsInEnrollmentType(
+      study.identifier,
+      token,
+      'enrolled',
+      false,
+    )
+    const dataWithoutTestMembers = resultEnrolled.items
+    if (tab === 'TEST') {
+      const resultAll = await ParticipantService.getAllParticipantsInEnrollmentType(
+        study.identifier,
+        token,
+        'all',
+        true,
+      )
+      const resultWithdrawn = await ParticipantService.getAllParticipantsInEnrollmentType(
+        study.identifier,
+        token,
+        'withdrawn',
+        false,
+      )
+      const dataAllTestMembers = resultAll.items
+      const dataWithdrawnMembers = resultWithdrawn.items
+      let testParticipants = dataAllTestMembers.filter(
```

Comment:
you don't need to use external ids. All participants have the id -- just filter on that